sdk/Trace/TracerProvider.php

```diff
@@ -4,60 +4,54 @@
 
 namespace OpenTelemetry\Sdk\Trace;
 
+use function is_array;
 use OpenTelemetry\Sdk\InstrumentationLibrary;
 use OpenTelemetry\Sdk\Resource\ResourceInfo;
 use OpenTelemetry\Sdk\Trace\Sampler\AlwaysOnSampler;
 use OpenTelemetry\Sdk\Trace\Sampler\ParentBased;
-use OpenTelemetry\Sdk\Trace\SpanProcessor\SpanMultiProcessor;
 use OpenTelemetry\Trace as API;
+use function register_shutdown_function;
 
 final class TracerProvider implements API\TracerProvider
 {
-    /**
-     * @var Tracer[]
-     */
-    protected $tracers;
-
-    /**
-     * @var IdGenerator
-     */
-    protected $idGenerator;
-
-    /**
-     * @var SpanMultiProcessor
-     */
-    protected $spanProcessors;
-
-    /**
-     * @var ResourceInfo
-     */
-    private $resource;
-
-    /**
-     * @var Sampler
-     */
-    private $sampler;
-
-    public function __construct(?ResourceInfo $resource = null, ?Sampler $sampler = null, ?IdGenerator $idGenerator = null)
-    {
-        $this->spanProcessors = new SpanMultiProcessor();
-        $this->resource = $resource ?? ResourceInfo::emptyResource();
-        $this->sampler = $sampler ?? new ParentBased(new AlwaysOnSampler());
-        $this->idGenerator = $idGenerator ?? new RandomIdGenerator();
+    /** @var array<string, API\Tracer> */
+    private $tracers;
+
+    /** @readonly */
+    private TracerSharedState $tracerSharedState;
+
+    /** @param list<SpanProcessor>|SpanProcessor $spanProcessors */
+    public function __construct(
+        $spanProcessors = [],
```

Comment:
I made it so you can pass 1 (or an array of) span processors to the constructor, versus needing to call `addSpanProcessor` a bunch of times. This is mainly needed so that `TracerSharedState` could be created with the full set of processors.